src/hooks/useRedirect.ts

```diff
@@ -0,0 +1,16 @@
+import React, { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { Domains, domains } from "@/router";
+
+const useRedirect = (domain: Domains, redirectCondition = true) => {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (redirectCondition) {
+      navigate(domains[domain]);
+    }
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    if (redirectCondition) {
      navigate(domains[domain]);
    }
  }, [domain, navigate, redirectCondition]);
```
의존성 배열이 누락된 것 같아요~

넣어줄 의존성들이 변하지 않는다고 확신할 수 있더라도 넣어주는걸 권장한다고 해요 [참고](https://github.com/facebook/react/issues/14920#issuecomment-470945051)

`eslint`의 `react-hooks/exhaustive-deps`가 작동하지 않는 것 같은데요..? 🤔